[ENG-39110] model driven commit parameters (#118)
NSO-developer/nso-gitbook: operation-and-usage/operations/lifecycle-operations.md

* `no-deploy`: Write service data without invoking the service create callback.
* `reconcile`: Reconcile service ownership for existing configuration.
  The `keep-non-service-config`, `discard-non-service-config`, `attach-non-service-config`, and `detach-non-service-config` leafs control how non-service-owned data is handled.
  The `include` and `exclude` leaf-lists limit reconciliation to selected service configuration paths.
* `use-lsa`, `no-lsa`: Control whether LSA nodes are handled as LSA nodes or as ordinary devices.
* `commit-queue`: Commit through the commit queue instead of pushing configuration transactionally in the same operation.
  The `async`, `sync`, and `bypass` leafs select the queue behavior.
  The `sync` container accepts either `timeout` or `infinity`.
  The `lock`, `block-others`, `atomic`, and `error-option` leafs control the resulting queue item.
  See [Commit Queue](nso-device-manager.md#user_guide.devicemanager.commit-queue) for details and error-recovery behavior.
* `trace-id`: Legacy trace identifier support as a commit parameter.
  Prefer Trace Context where available.

> The model enforces parameter combinations. For example, `dry-run` cannot be combined with `no-overwrite`, `no-out-of-sync-check`, or `commit-queue`, and `use-lsa` cannot be combined with `no-lsa`.

Reviewer: I first read this as some combinations being required. suggest to reword:

```suggestion
Some combinations of parameters are not allowed. For example, `dry-run` cannot be combined with `no-overwrite`, `no-out-of-sync-check`, or `commit-queue`, and `use-lsa` cannot be combined with `no-lsa`.
```
